### DISCUSSION TITLE
How do you guys communicate with servers from your apps?

### DISCUSSION THREAD

**Discussion by mrfrasier** [airsdk/Adobe-Runtime-Support#3250] [General]
Over the years we've used a number of backends to allow our apps to communicate with servers. NodeJS, REST endpoints, and probably our favorite was the now-defunct AMF PHP for how easy it made things using the Action Message Format. The Harman team is now building in web sockets and who knows what other goodies are in store.

What's your favorite way of app-to-server comms?

**Comment by ghost** [comment]
My choice is Back4app SaaS. You can manage without additional ANEs, write backend code in js, handle realtime live queries through WebSocket, push notifications and so on.
